.github/workflows/publish-release.yml

```diff
@@ -70,32 +66,31 @@ jobs:
         run: |
           yarn install --immutable
           yarn build
-      - uses: actions/cache@v4
-        id: restore-build
+      - name: Upload build artifacts
+        uses: actions/upload-artifact@v4
         with:
+          name: publish-release-artifacts-${{ github.sha }}
+          include-hidden-files: true
+          retention-days: 4
           path: |
             ./packages/**/dist
-            ./node_modules
-          key: ${{ github.sha }}
+            ./node_modules/.yarn-state.yml
 
   npm-publish-dry-run:
     name: Publish to NPM (dry run)
     runs-on: ubuntu-latest
     needs: publish-release
     steps:
-      - uses: actions/checkout@v4
+      - name: Checkout and setup environment
+        uses: MetaMask/action-checkout-and-setup@v1
         with:
+          is-high-risk-environment: true
           ref: ${{ github.sha }}
-      - uses: actions/cache@v4
-        id: restore-build
+      - name: Restore build artifacts
+        uses: actions/download-artifact@v4
         with:
-          path: |
-            ./packages/**/dist
-            ./node_modules
-          key: ${{ github.sha }}
-          fail-on-cache-miss: true
-      - run: npm config set ignore-scripts true
```

Comment:
Should we not keep this?

Reply:
We don't use `npm` for publishing, I think it's just a leftover from an old workflow. It's not in the [`MetaMask/core`](https://github.com/MetaMask/core/blob/288c82b5d2c5f0423320b7ff2b5626e3d44082f7/.github/workflows/publish-release.yml) workflow either for example.